lib/model/model_test.go

```diff
@@ -1768,6 +1768,23 @@ func TestIgnores(t *testing.T) {
 	changeIgnores(t, m, []string{})
 }
 
+func waitForState(t *testing.T, m *model, status string) {
+	t.Helper()
+	timeout := time.Now().Add(2 * time.Second)
+	var err error
+	for !time.Now().After(timeout) {
+		_, _, err = m.State("default")
+		if err == nil && status == "" {
+			return
+		}
+		if err != nil && err.Error() == status {
+			return
+		}
+		time.Sleep(10 * time.Millisecond)
+	}
+	t.Fatalf("Timed out waiting for status: %s, current status: %v", status, err)
+}
+
```

Comment:
Perhaps should take a folder name to be more generally useful?
Also, perhaps all of these should start going into some sort of model_test_utils.go or something

Reply:
Agreed about the separate file for utilities, but that's a potentially deep rabbithole so I'd rather not do it in right now in this PR.